Source/Core/DolphinQt/ScriptManager.cpp

```diff
@@ -0,0 +1,228 @@
+{
+  auto* layout = new QGridLayout();
+  m_load_script_button = new NonDefaultQPushButton(tr("Load Script"));
+
+  m_play_or_stop_script_button = new NonDefaultQPushButton(tr("N/A"));
+  m_play_or_stop_script_button->setStyleSheet(tr("background-color:grey;"));
+
+  script_name_list_widget_ptr = new QListWidget;
+  script_name_list_widget_ptr->setMinimumHeight(100);
+  script_name_list_widget_ptr->setMaximumHeight(200);
+  script_name_list_widget_ptr->setMinimumWidth(550);
+  script_name_list_widget_ptr->setMaximumWidth(800);
+
+  QLabel* script_name_header = new QLabel(tr("Script Name:"));
+  auto* script_name_box = new QGridLayout();
+  script_name_box->addWidget(script_name_header, 0, 0, Qt::AlignTop);
+  script_name_box->addWidget(script_name_list_widget_ptr, 1, 0, Qt::AlignTop);
+
+  QLabel* output_header = new QLabel(tr("Output:"));
+  script_output_list_widget_ptr = new QListWidget;
+  script_output_list_widget_ptr->setMinimumHeight(300);
+  script_output_list_widget_ptr->setMaximumHeight(600);
+  script_output_list_widget_ptr->setMinimumWidth(550);
+  script_output_list_widget_ptr->setMaximumWidth(800);
+  auto* output_box = new QGridLayout();
+  output_box->addWidget(output_header, 0, 0, Qt::AlignTop);
+  output_box->addWidget(script_output_list_widget_ptr, 1, 0, Qt::AlignTop);
+
+  script_output_list_widget_ptr->setSpacing(1);
+  script_output_list_widget_ptr->setStyleSheet(tr("background-color:white;"));
+
+  layout->addWidget(m_load_script_button, 0, 0, Qt::AlignTop);
+  layout->addWidget(m_play_or_stop_script_button, 0, 1, Qt::AlignTop);
+  layout->addLayout(script_name_box, 1, 0, Qt::AlignTop);
+  layout->addLayout(output_box, 1, 1, Qt::AlignTop);
+  setWindowTitle(tr("Script Manager"));
+  setLayout(layout);
+}
+
+void ScriptManager::ConnectWidgets()
+{
+  connect(m_load_script_button, &QPushButton::clicked, this, &ScriptManager::LoadScriptFunction);
+  connect(m_play_or_stop_script_button, &QPushButton::clicked, this,
+          &ScriptManager::PlayOrStopScriptFunction);
+  connect(script_name_list_widget_ptr, &QListWidget::itemSelectionChanged, this,
+          &ScriptManager::UpdateButtonText);
+}
+
+void ScriptManager::LoadScriptFunction()
+{
+  std::lock_guard<std::mutex> lock(script_start_or_stop_lock);
+  auto& settings = Settings::Instance().GetQSettings();
+  QString path = DolphinFileDialog::getOpenFileName(
+      this, tr("Select a File"),
+      settings.value(QStringLiteral("scriptmanager/lastdir"), QString{}).toString(),
+      QStringLiteral("%1 (*.lua *.txt *.py);;%3 (*)")
+          .arg(tr("All Lua/txt files"))
+          .arg(tr("All Python files"))
+          .arg(tr("All Files")));
+
+  if (!path.isEmpty())
+  {
+    settings.setValue(QStringLiteral("scriptmanager/lastdir"),
+                      QFileInfo(path).absoluteDir().absolutePath());
+    row_num_to_is_running[next_unique_identifier] = false;
+    QStringList list;
+    list << path;
+    script_name_list_widget_ptr->addItem(path);
```

Comment:
```suggestion
    script_name_list_widget_ptr->addItem(QStringList(path));
```

is more compact compared to constructing the list and manually appending to it.